### DISCUSSION TITLE
Help in building sitemap.xml appropriately

### DISCUSSION THREAD

**Discussion by dekellevy93** [patrickdemooij9/SeoToolkit.Umbraco#237] [Q&A]
Hi guys!
i wonder what's seems to be the problem in building the sitemap.xml in my site.
the first assumption is i need the my site domain to be https:
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/e1d6da99-9d7e-4d30-9734-ce30a21246c2)
so all the canonical urls of my pages be https as well:
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/0b91f781-a725-4ef4-ad8c-a3ff9b28e47f)
but when i do it, it can't seems to generate sitemap.xml appropriately:
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/532be193-4d84-4a61-b162-60595d843f72)
when i change the domain to http, its building the sitemap, but with http prefix(of course), and even if im entering the canonical url by hand.
its just seems like a specific site has a base url which is the only one that works  for him, and when the sitemap is trying to build with different url it doesn't build it. for example it works fine in my local environment:
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/c91df4c1-106e-4653-b35d-b39f2881f4ca)
it build with https and everything:
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/73b73f1b-6d99-4cd4-ae8f-3272f789fb15)
what could be the problem?
Thanks for advance,
Dekel.

**Comment by patrickdemooij9** [comment]
Hi @dekellevy93 
The canonical URL doesn't have an impact on the sitemap generation, they are completely separate.

For some reason, Umbraco is not able to match the URL you are visiting to the URL you have given up as domain. Are you sure that you are visiting the same URL as the one in the domain field? Also, in the sitemap index that it is currently generating, is that generating the same url as that url that you are visiting?

**Reply by dekellevy93** [reply]
Hi @patrickdemooij9 , thank you for your reply!
yeah, i'm visiting the same url i used in the domain field + i inserted this url to  "WebRouting": { "UmbracoApplicationUrl": } attribute in the appsettings file.
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/b3beb950-bc3d-4dcc-aa3d-428b5bf1830a)
the only node he is building when im using https is the sitemap node..

**Reply by patrickdemooij9** [reply]
Hmm oke, that is strange. I'll have to see if I can reproduce this issue myself

**Reply by dekellevy93** [reply]
@patrickdemooij9 hi Patrick, any news?

**Reply by patrickdemooij9** [reply]
Hi @dekellevy93 
Sorry for the late reply, I haven't quite figured out what is going wrong here. Could you perhaps do one more thing? I just looked at the domain code and I think it might cause this if you have a domain linked to an unpublished item.

Could you run this code on production in a template?
`<p>@UmbracoContext.Domains.GetAll(false).Count()</p>`
If this shows more than 1, then there is another domain hidden somewhere (which would explain this behaviour).

If this doesn't work, then I will see if I can fix it through some other way

**Reply by dekellevy93** [reply]
@patrickdemooij9 i checked it on my staging environment, and im getting one domain.
which is supposed to be https:
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/de936df2-ac79-4f28-b3d4-e8e4eb903914)
also when I run health check on the security group
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/0ae7862a-4edc-4e0b-816d-0787adb0fec7)
but still it wasn't able to build the sitemap:
![image](https://github.com/patrickdemooij9/SeoToolkit.Umbraco/assets/58995464/b6505f82-c629-4138-8b12-d0eabb6823da)
Thanks,
Dekel.

**Reply by patrickdemooij9** [reply]
Hi @dekellevy93 
I just pushed a new release that should (hopefully) fix this for you. Could you take a look and see if it works?
